### ISSUE TITLE
Calling Async Function in Sync Function

### ISSUE THREAD

**Issue by Pebaz** [not-fl3/macroquad#184]
Is there any possible way that a function could be added so that functions like `load_file()` can be called in a normal Rust function? I don't actually need it to be async at all, just need it to work (blocking is good).

I need to be able to do this:

```rust
fn function_exposed_to_scripting_language_without_async_support()
{
    let future = load_file("something.rhai");

    // Need this:
    let result = block_on(future);
}
```

I tried to use the "futures" library on WASM and it broke the JavaScript somehow. If there's another WASM-capable futures library I haven't been able to find it.

**Comment by not-fl3** [comment]
No, this is not possible due to browsers design - JS/Wasm functions should be non-blocking, there is no way to block in the browser. That browser limitation is the reason async/await was used in the first place: https://github.com/not-fl3/macroquad#asyncawait

**Comment by oli-obk** [comment]
that said, I believe I could implement a light-weight version of https://crates.io/crates/wasm-bindgen-futures to allow you to return futures from js to Rust
